src/typescriptGenerator.ts

```diff
@@ -76,22 +77,34 @@ export async function generateTypeScriptLibrary(
 
   const shouldGenerateLicense: boolean =
     (await host.GetValue("license-header")) || false;
-
   const hideClients: boolean = (await host.GetValue("hide-clients")) || false;
+  const armLibrary: boolean =
+    (await host.GetValue("azure-arm")) ||
+    (await host.GetValue("openapi-type")) === "arm" ||
+    false;
+  const ignoreNullableOnOptional: boolean =

```

Comment:
I think you can do the following and later on you just need to check for `ignoreNullableOnOptional` without also checking for armLibrary

```typescript
const ignoreNullableOnOptional: boolean = (await host.GetValue("ignore-nullable-on-optional")) ??  (await host.GetValue("azure-arm")) 
```